src/Castle.Core/NullStateStaticAnalysisAttributes.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright 2004-2023 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+	using System;
+
+#if !NETSTANDARD2_1_OR_GREATER && !NETCOREAPP3_0_OR_GREATER
+	[AttributeUsage(AttributeTargets.Method, Inherited = false)]
+	internal sealed class DoesNotReturnAttribute : Attribute
```

Comment:
I haven't yet verified whether null-state static analysis will work correctly when these polyfill attributes are declared `internal`. (I am fairly certain though that they should _not_ be declared `public`!)

Reply:
Done. Verified that declaring the polyfill attributes `internal` works just fine.